### PR TITLE
Test: refactor liquidation validations and solve lint warnings

### PR DESCRIPTION
**Pull request by ivansolomonoff**
## Security Checklist

- [ ] 1. Re-Entrancy
- [ ] 2. Arithmetic Over/Under Flows
- [ ] 3. Unexpected Ether
- [ ] 4. Delegatecall
- [ ] 5. Default Visibilities
- [ ] 6. Entropy Illusion
- [ ] 7. External Contract Referencing
- [ ] 8. Short Address/Parameter Attack (off chain)
- [ ] 9. Unchecked CALL Return Values
- [ ] 10. Race Conditions / Front Running
- [ ] 11. Denial Of Service (DOS)
- [ ] 12. Block Timestamp Manipulation
- [ ] 13. Constructors with Care
- [ ] 14. Uninitialized Storage Pointers
- [ ] 15. Floating Points and Precision
- [ ] 16. Tx.Origin Authentication
- [ ] 17. Address.isContract Re-Entrancy via Constructor

### ⚠️ NOTES ⚠️

Make sure to think about each of these exploits in this PR.

### CODE REVIEW

**Comment by GopherJ** [comment]
@ivansolomonoff might be cool to also clear our warnings
image

File: test-suites/helpers/validated-steps.ts [hunk 1]
```diff
@@ -567,22 +595,258 @@ const checkBeforeLiquidation = async (before: LiquidationValidationData) => {
...
+  after: AfterLiquidationValidationData
 ) => {
-  return true;
+  // borrower's Token balance is the same
+  expect(after.borrowerTargetTokenBalance).equal(
+    before.borrowerTargetTokenBalance
+  );
+
+  // borrower's pToken balance is subtracted amountToLiquidate+liq bonus (bonus is paid by borrower)
+  assertAlmostEqual(
```
Comment (yrong) [inline test-suites/helpers/validated-steps.ts:613]: will this assert work without consideration of price of each asset?

File: test-suites/helpers/validated-steps.ts [hunk 2]
```diff
@@ -567,22 +595,258 @@ const checkBeforeLiquidation = async (before: LiquidationValidationData) => {
...
+    after.borrowerTargetXTokenBalance,
+    before.borrowerTargetXTokenBalance.sub(
+      amountToLiquidate.mul(before.liquidationBonus).div(10000)
+    )
+  );
```
Comment (yrong) [inline test-suites/helpers/validated-steps.ts:618]: will this assert work without consideration of price of each asset?

File: test-suites/helpers/validated-steps.ts [hunk 3]
```diff
@@ -567,22 +595,258 @@ const checkBeforeLiquidation = async (before: LiquidationValidationData) => {
...
+      after.liquidatorTargetTokenBalance,
+      before.liquidatorTargetTokenBalance.sub(amountToLiquidate)
+    );
+    // liquidator's pToken balance is incremented in liquidatedAmount+bonus (plus accrued interest)
+    assertAlmostEqual(
+      after.liquidatorTargetXTokenBalance,
+      before.liquidatorTargetXTokenBalance.add(
+        amountToLiquidate.mul(before.liquidationBonus).div(10000)
+      )
+    );
```
Comment (yrong) [inline test-suites/helpers/validated-steps.ts:638]: also these checks will work?

Comment (yrong) [reply test-suites/helpers/validated-steps.ts:638]: would suggest we make sure every check make sense if in a common function to reuse. or we just wrap result including enough info and let test verify by itself.

**Comment by ivansolomonoff** [comment]
> @ivansolomonoff might be cool to also clear our warnings image



> @ivansolomonoff might be cool to also clear our warnings image

yup, it's done now.

File: test-suites/helpers/validated-steps.ts (hunk 2 above)
Comment (ivansolomonoff) [reply test-suites/helpers/validated-steps.ts]: yeah, I need to recheck all the logic, I put everything together so it should be simpler now. Let you know when it's done.

Comment (ivansolomonoff) [reply test-suites/helpers/validated-steps.ts]: I reviewed all validations, should be ok now.

**Review by GopherJ — APPROVED** [review]

**Comment by yrong** [comment]
https://github.com/para-space/paraspace-core/pull/113/commits/73a672d5fbb0ae3d2aeef5486b9dd2d956eb05ec
@ivansolomonoff  some minor polish here.  for non-borrowed case maybe we do not need it after double confirm with team so just remove it from the common validation logic here.

**Pull request merged**